### PR TITLE
Fix broken layout in banner.mdx

### PR DESCRIPTION
**Pull request by patrickhlauke**
Adds wrapping `<div>` to fix the broken layout for the https://primer.style/components/banner#dismissable section in the documentation.

screenshot of the currently broken documentation section, showing content awkwardly split over four columns when it should only be two columns

### CODE REVIEW

File: content/components/banner.mdx
Comment (patrickhlauke) [inline content/components/banner.mdx:177]:
```suggestion
Use the following as a guideline:

```

**Review by keithamus — APPROVED** [review]

**Comment by patrickhlauke** [comment]
I'm unsure why the preview/build step is failing here (as I only took inspiration from the following section in the doc that used the approach of wrapping things in a `<div>` - if anybody knows/can fix this, I'd be grateful /cc @keithamus

**Comment by patrickhlauke** [comment]
Does it need to have the unordered list written out fully as HTML as well, to work around this?

File: content/components/banner.mdx
Comment (patrickhlauke) [inline content/components/banner.mdx:179]:
```suggestion
<ul>
  <li>If you're on a page in which you can do other tasks, and the banner is not blocking / does not have to be addressed, then you should be able to have a close button.</li>
```

File: content/components/banner.mdx
Comment (patrickhlauke) [inline content/components/banner.mdx:180]:
```suggestion
  <li>If accidental closure of the banner would leave the user confused or missing something critical, it should not have a close button.</li>
</ul>
```

**Review by keithamus — APPROVED** [review]
Strange that it needed that... but sure, still LGTM :+1:

**Comment by patrickhlauke** [comment]
Some shortcoming in the markdown converter it seems, but yes confirming that visually the preview now looks good https://primer-8e64b013b2-26441320.drafts.github.io/components/banner#dismissable

Merging (I'm assuming it's ok for me to merge here once it's been approved - if not, please let me know if there's a more official process to follow)

**Pull request merged**
